packages/plans-grid-next/src/components/shared/storage/components/storage-dropdown.tsx

```diff
@@ -93,52 +93,48 @@ const StorageDropdown = ( {
 		( select ) => select( WpcomPlansUI.store ).getSelectedStorageOptionForPlan( planSlug, siteId ),
 		[ planSlug, siteId ]
 	);
-	const defaultStorageOption = useDefaultStorageOption( { planSlug } );
+	const defaultStorageOptionSlug = useDefaultStorageOption( { planSlug } );
 	const availableStorageAddOns = AddOns.useAvailableStorageAddOns( { siteId } );
 	const planStorage = usePlanStorage( planSlug );
 
 	useEffect( () => {
 		if ( ! selectedStorageOptionForPlan ) {
-			defaultStorageOption &&
+			defaultStorageOptionSlug &&
 				setSelectedStorageOptionForPlan( {
-					addOnSlug: defaultStorageOption,
+					addOnSlug: defaultStorageOptionSlug,
 					planSlug,
 					siteId,
 				} );
 		}
 	}, [
-		defaultStorageOption,
+		defaultStorageOptionSlug,
 		planSlug,
 		selectedStorageOptionForPlan,
 		setSelectedStorageOptionForPlan,
 		siteId,
 	] );
 
-	const defaultStorageItem = useMemo(
-		() => ( {
-			key: defaultStorageOption || '',
-			name: (
-				<StorageDropdownOption price="" totalStorage={ planStorage } />
-			 ) as unknown as string,
-		} ),
-		[ defaultStorageOption, planStorage ]
-	);
+	const selectControlOptions = useMemo( () => {
+		// Get the default storage add-on meta or the storage included with the plan
+		const defaultStorageAddOnMeta = getSelectedStorageAddOn(
+			storageAddOns,
+			defaultStorageOptionSlug || ''
+		) || { addOnSlug: defaultStorageOptionSlug, prices: null, quantity: 0 };
 
-	const selectControlOptions = [ defaultStorageItem ].concat(
-		availableStorageAddOns?.map( ( addOn ) => {
-			const addOnStorage = addOn.quantity ?? 0;
+		return [ defaultStorageAddOnMeta, ...availableStorageAddOns ]?.map( ( addOn ) => {
```

Comment:
`NB:`

A little confusing destructuring `AddOnMeta` with `AddOns`. Not a strong opinion, but thoughts about renaming to `defaultStorageAddOnMeta` to `defaultStorageAddOn`?

Reply:
Hmm, I named it because `getSelectedStorageAddOn` returns `AddOnMeta`. 🤔 It is a bit confusing because the value of `defaultStorageAddOnMeta` can also be `{ addOnSlug: defaultStorageOptionSlug, prices: null, quantity: 0 }`, which is clearly not `AddOnMeta`. I'll try to refactor this to make it clearer.

Reply:
Done via https://github.com/Automattic/wp-calypso/pull/99310/commits/4cfe2873ecbc968df5c30e7a9befbb3ca27217f8! Hope it's clearer now.